### PR TITLE
[go1.15] Update to go1.15.7

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**

/kind feature
/area dependency

**What this PR does / why we need it**:

- Update to go1.15.7

Tracking issue: https://github.com/kubernetes/release/issues/1851
Slack thread: https://kubernetes.slack.com/archives/C2C40FMNF/p1610736817029800

cc: @kubernetes/release-engineering 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- Kubernetes is now built using go1.15.7
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
For RelEng:
/lgtm
/approve

/assign @dims

**Comment by cpanato** [comment]
pull-kubernetes-verify failed and checking the logs looks like we need to update somethings

**Comment by cpanato** [comment]
@dims and @justaugustus the `pull-kubernetes-verify` job complained and ask to run some hack scripts to update some files. PTAL

**Comment by justaugustus** [comment]
/assign @liggitt 
(for `staging/` changes)

File: staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go [hunk 1]
```diff
@@ -342,13 +342,13 @@ func init() {
 }
 
 var fileDescriptor_aaac5994f79683e8 = []byte{
-	// 1104 bytes of a gzipped FileDescriptorProto
+	// 1102 bytes of a gzipped FileDescriptorProto
```
Comment (liggitt) [inline staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: whoa... didn't expect these generation changes... any idea what happened here?

Comment (cpanato) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: to be honest no 😄

Comment (liggitt) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: hmm... did we accidentally bump proto generator libraries somehow in one of the build images?

Comment (cpanato) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: I will go over the PRs, but I don't remember bumping this

for reference:
- https://github.com/kubernetes/release/pull/1857
- https://github.com/kubernetes/repo-infra/pull/223

Comment (justaugustus) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: > hmm... did we accidentally bump proto generator libraries somehow in one of the build images?

Potentially a result of this PR to enable more arches for kube-cross: https://github.com/kubernetes/release/pull/1853

cc: @dims

Comment (dims) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: @liggitt @justaugustus yep, that's the one. how bad is it?

File: staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go [hunk 2]
```diff
@@ -363,7 +363,7 @@ func (m *TopologyInfo) GetNodes() []*NUMANode {
 
 // NUMA representation of NUMA node
 type NUMANode struct {
-	ID                   int64    `protobuf:"varint,1,opt,name=ID,json=iD,proto3" json:"ID,omitempty"`
+	ID                   int64    `protobuf:"varint,1,opt,name=ID,proto3" json:"ID,omitempty"`
```
Comment (liggitt) [inline staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: would like an ack from @kubernetes/sig-node-api-reviews this doesn't impact any pod resource API serialization

File: staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/admissionregistration/v1/generated.pb.go:345]: ok, that explains the diff. would like an ack from node API reviewers that the change in the generated protobuf tag to drop json field name doesn't affect their use

File: staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go (hunk 2 above)
Comment (dims) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: @SergeyKanzhelev @ehashman Can you please ack this change?

Comment (fromanirh) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: I think @AlexeyPerevalov can help here! I also contributed to the feature, digging into the git history to restore the full context.

Comment (SergeyKanzhelev) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: Just to be clear, there is the same change on the`type Device struct` structure in two files above.

Comment (fromanirh) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: My take, but please let's wait for @AlexeyPerevalov 's to confirm, is that we don't _need_ anything special for `NUMANode`.  It should behave like any other `podresources` API message. Not sure if some setting got lost somewhere. The intent was to just do what the rest of the proto file did. I'll research which settings `podresources` originally used. Also, thinking about it @RenaudWasTaken took care of the v1alpha1 -> v1 promotion, so he may want to comment here? I'm not sure something changed when moving to v1.

Comment (liggitt) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: if nothing in the kubelet path uses json proto serialization,  then this looks like a no-op change to me

Comment (AlexeyPerevalov) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: It didn't impact.
podresources is not only the one place, where I found json=iD. I also found it in libopenstorage an d deviceplugin. Probably, it's due to generator modifies the first letter (to lower case)

Comment (liggitt) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: right. none of those paths use json proto serialization, correct?

Comment (RenaudWasTaken) [reply staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.pb.go:366]: > right. none of those paths use json proto serialization, correct?

AFAIK nothing uses the json proto

**Comment by mrunalp** [comment]
Needs rebase

**Comment by liggitt** [comment]
/approve

needs rebase/regen

**Comment by cpanato** [comment]
rebased @liggitt, PTAL

**Pull request merged**